### PR TITLE
identifier tweaks

### PR DESCRIPTION
**Pull request by bnewbold**
The motivation was to take some of the small proposed changes from identifier specification work and put them in PR form for review here.

The NSID "name" component length change (from 128 to 63): i'm actually now kind of ambivalent after implementing this. One of the motivations was realizing huge names would bloat AT-URIs. But maybe it just doesn't matter!

NSID other restrictions: I think tracking handles to domains very closely has been a good move. I'm not sure the same needs to apply to NSIDs, but it is weird to have extra rules around NSID *domain authority* section other than "valid hostname in reverse order". I am in favor of locking down the "name" section very harshly (to just ASCII letters, either case). We can allow trailing digits or whatever later if we want, once we verify which programming languages do/don't allow that in variable names.

### CODE REVIEW

**Comment by bnewbold** [comment]
The tests are failing due to a test NSID named `io.example.stream1` (which would be disallowed by the NSID syntax changes).

Let's review/discuss the syntax changes first, I can clean up all the tests to match our decisions after.

File: packages/did-resolver/src/web-resolver.ts [hunk 1]
```diff
@@ -20,7 +20,9 @@ export class DidWebResolver extends BaseResolver {
     } else if (parts.length === 1) {
       path = parts[0] + DOC_PATH
     } else {
-      path = parts.join('/') + '/did.json'
+      // how we *would* resolve a did:web with path, atproto supported it
+      //path = parts.join('/') + '/did.json'
+      throw new UnsupportedDidWebPathError(did)
```
Comment (dholms) [inline packages/did-resolver/src/web-resolver.ts:25]: yup 👍

File: packages/identifier/src/did.ts [hunk 2]
```diff
@@ -38,7 +38,7 @@ export const ensureValidDid = (did: string): void => {
     throw new InvalidDidError('DID can not end with ":" or "%"')
   }
 
-  if (did.length > 8 * 1024) {
+  if (did.length > 2 * 1024) {
     throw new InvalidDidError('DID is far too long')
```
Comment (dholms) [inline packages/identifier/src/did.ts:42]:
```suggestion
    throw new InvalidDidError('DID is far long')
```

**Comment by AaronGoldman** [comment]
ASCII can be combined with a recommendation for how to get Unicode if that is a thing you want to do.

https://en.wikipedia.org/wiki/Punycode

**Comment by AaronGoldman** [comment]
ASCII can be combined with a recomendaton for 

https://en.wikipedia.org/wiki/Punycode

File: packages/did-resolver/src/web-resolver.ts [hunk 3]
```diff
@@ -20,7 +20,9 @@ export class DidWebResolver extends BaseResolver {
     } else if (parts.length === 1) {
       path = parts[0] + DOC_PATH
     } else {
-      path = parts.join('/') + '/did.json'
+      // how we *would* resolve a did:web with path, atproto supported it
```
Comment (dholms) [inline packages/did-resolver/src/web-resolver.ts:23]:
```suggestion
      // how we *would* resolve a did:web with path, if atproto supported it
```

File: packages/identifier/src/did.ts (hunk 2 above)
Comment (dholms) [inline packages/identifier/src/did.ts:42]: since we've limited to 2, can probably just change the error text to "too long" instead of "far too long" 😅

Comment (Matrix89) [reply packages/identifier/src/did.ts:42]: Since other errors return what's valid/invalid, shouldn't this error also mention the max length?

**Comment by bnewbold** [comment]
@dholms made small changes to get CI to pass:

* rebased
* incorporated small review fixes
* fixed errors, including some from rebasing
* renamed NSIDs in a few package tests to not include digits (just spelled out numbers)

No behavioral changes. I plan to merge this later tonight or tomorrow.

**Review by dholms — APPROVED** [review]
yup looks good

feel free to merge whenevs 👌

**Pull request merged**